### ISSUE TITLE
viewed chapters

### ISSUE THREAD

**Issue by jdxcode** [rust-lang/mdBook#1085]
I'd love it if there was a way for me to tell how far along a book I am. If previously viewed chapters showed up gray or maybe a little "unread" label showed on unread chapters?

Of course I could use bookmarks but that would require me to have to update the bookmark every time I view a new chapter.

Open to ideas and happy to implement as well.

**Comment by weihanglo** [comment]
Great! I am thinking of the same feature these days and already have some open questions:

- Would it enable by default? If yes, does it break current user experience? I know some of the book in the wild are used as references but not tutorials. It should be disabled on these types of books.
- What the definition of _read_ behavior? Scroll down to a specific percentage like 80%? Or just count the click?
- Supposed the progress of each chapter would be stored in browsers `localstorage` or something like that. But what if the book structure gets some changes? Shall we maintain the structure on our own or just invalidate oudated progress?

**Comment by jdxcode** [comment]
I don't think you need to track anything, the browser already tracks viewed pages so we just need to add a style for `a:visited` like I showed: https://www.w3schools.com/cssref/sel_visited.asp

**Comment by weihanglo** [comment]
My fault. Shouldn't expand the question too much. Just make it simple and use `visited` selector.

**Comment by jdxcode** [comment]
no worries, I was considering the same exact thing until it dawned on me this is a standard browser thing :)

In regards to enabling by default I'm not sure. I imagine we could just have a css class to enable/disable the functionality.
